homeassistant/components/config/device_registry.py

```diff
@@ -50,7 +51,9 @@
     registry = await async_get_registry(hass)
 
     entry = registry.async_update_device(
-        msg['device_id'], area_id=msg['area_id'])
+        msg['device_id'],
+        area_id=msg['area_id'],
```

Comment:
We should do this differently, because these keys might not be defined (existing bug)

I am a big fan of how the shopping list does it. Just drop the keys that are not changes and then use `**msg`:

https://github.com/home-assistant/home-assistant/blob/8b38b82e73d8ce83b3bd6200d2fd559d39c19e87/homeassistant/components/shopping_list/__init__.py#L327-L330

Reply:
Something like this??